script.module.clouddrive.common/clouddrive/common/service/player.py

```diff
@@ -60,27 +62,91 @@ def stop(self):
 
 class KodiPlayer(KodiUtils.kodi_player_class()):
     def __init__(self, *args):
-        pass
```

Comment:
Generally it is a bad practice to set variables on the home window. Can you use another alternative such as an hidden setting or a property on the listitem itself?

Reply:
I'm using the home window because I didn't found a better alternative. A hidden setting would make persistent a value that I need for a short period of time only. Making it permanent in the settings will complicate the functionality.
A property in the listitem would be nice if worked, but it didn't. I set the property before calling setResolvedUrl and then try to read it with the infolabel ListItem.Property() when the playback has started, but it always returns empty. I'm guessing the listitem was changed for another instance at playing time.
So the property in the home window is the only solution it's working for me right now.